### ISSUE TITLE
Cannot install lem.

### ISSUE THREAD

**Issue by aykaramba** [cxxxr/lem#313]
Hi.
When installing lem i get the following error:

Installing from github cxxxr/lem
To load "lem":
  Load 1 ASDF system:
    lem
; Loading "lem"
To load "inquisitor":
  Load 3 ASDF systems:
    alexandria anaphora asdf
  Install 1 Quicklisp release:
    inquisitor
Downloading http://beta.quicklisp.org/archive/inquisitor/2018-02-28/inquisitor-20180228-git.tgz
##########################################################################
; Loading "inquisitor"
.
; Loading "lem"
To load "yason":
  Load 2 ASDF systems:
    alexandria trivial-gray-streams
  Install 1 Quicklisp release:
    yason
Downloading http://beta.quicklisp.org/archive/yason/2016-02-08/yason-v0.7.6.tgz
##########################################################################
; Loading "yason"

; Loading "lem"
To load "esrap":
  Load 2 ASDF systems:
    alexandria asdf
  Install 1 Quicklisp release:
    esrap
Downloading http://beta.quicklisp.org/archive/esrap/2018-04-30/esrap-20180430-git.tgz
##########################################################################
; Loading "esrap"

; Loading "lem"
To load "usocket":
  Load 1 ASDF system:
    split-sequence
  Install 1 Quicklisp release:
    usocket
Downloading http://beta.quicklisp.org/archive/usocket/2016-10-31/usocket-0.7.0.1.tgz
##########################################################################
; Loading "usocket"
[package usocket]..........
; Loading "lem"
To load "optima":
  Load 2 ASDF systems:
    alexandria closer-mop
  Install 1 Quicklisp release:
    optima
Downloading http://beta.quicklisp.org/archive/optima/2015-07-09/optima-20150709-git.tgz
##########################################################################
; Loading "optima"

; Loading "lem"
.[1/3] System 'lem' found. Loading the system..Aborted during step [1/3].
Unhandled UNDEFINED-FUNCTION in thread #<SB-THREAD:THREAD "main thread" RUNNING
                                          {10005505B3}>:
  The function ASDF/INTERFACE::OPERATION-FORCED is undefined.

Backtrace for: #<SB-THREAD:THREAD "main thread" RUNNING {10005505B3}>
0: ("undefined function" #<ASDF/LISP-ACTION:LOAD-OP >)
1: ((:METHOD ASDF/ACTION:PERFORM (ASDF/LISP-ACTION:LOAD-OP SWANK-LOADER::SWANK-LOADER-FILE)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
2: ((SB-PCL::EMF ASDF/ACTION:PERFORM) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">)
3: ((LAMBDA NIL :IN ASDF/ACTION:CALL-WHILE-VISITING-ACTION))
4: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS (ASDF/LISP-ACTION:LOAD-OP ASDF/LISP-ACTION:CL-SOURCE-FILE)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
5: ((:METHOD ASDF/ACTION:PERFORM-WITH-RESTARTS :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP > #<SWANK-LOADER::SWANK-LOADER-FILE "swank" "swank-loader">) [fast-method]
6: ((:METHOD ASDF/PLAN:PERFORM-PLAN (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1003E92213}>) [fast-method]
7: ((FLET SB-C::WITH-IT :IN SB-C::%WITH-COMPILATION-UNIT))
8: ((:METHOD ASDF/PLAN:PERFORM-PLAN :AROUND (T)) #<ASDF/PLAN:SEQUENTIAL-PLAN {1003E92213}>) [fast-method]
9: ((:METHOD ASDF/OPERATE:OPERATE (ASDF/OPERATION:OPERATION ASDF/COMPONENT:COMPONENT)) #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "lem"> :PLAN-CLASS NIL :PLAN-OPTIONS NIL) [fast-method]
10: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "lem"> :VERBOSE NIL)
11: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
12: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) #<ASDF/LISP-ACTION:LOAD-OP > #<ASDF/SYSTEM:SYSTEM "lem"> :VERBOSE NIL) [fast-method]
13: ((SB-PCL::EMF ASDF/OPERATE:OPERATE) #<unused argument> #<unused argument> ASDF/LISP-ACTION:LOAD-OP "lem" :VERBOSE NIL)
14: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
15: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "lem" :VERBOSE NIL) [fast-method]
16: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<CLOSURE (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1003E8A9CB}> :OVERRIDE T :KEY NIL :OVERRIDE-CACHE T :OVERRIDE-FORCING NIL)
17: ((LAMBDA NIL :IN ASDF/OPERATE:OPERATE))
18: (ASDF/SESSION:CALL-WITH-ASDF-SESSION #<CLOSURE (LAMBDA NIL :IN ASDF/OPERATE:OPERATE) {1003E850AB}> :OVERRIDE NIL :KEY NIL :OVERRIDE-CACHE NIL :OVERRIDE-FORCING NIL)
19: ((:METHOD ASDF/OPERATE:OPERATE :AROUND (T T)) ASDF/LISP-ACTION:LOAD-OP "lem" :VERBOSE NIL) [fast-method]
20: (ASDF/OPERATE:LOAD-SYSTEM "lem" :VERBOSE NIL)
21: (QUICKLISP-CLIENT::CALL-WITH-MACROEXPAND-PROGRESS #<CLOSURE (LAMBDA NIL :IN QUICKLISP-CLIENT::APPLY-LOAD-STRATEGY) {1003E8498B}>)
22: (QUICKLISP-CLIENT::AUTOLOAD-SYSTEM-AND-DEPENDENCIES "lem" :PROMPT NIL)
23: ((:METHOD QL-IMPL-UTIL::%CALL-WITH-QUIET-COMPILATION (T T)) #<unused argument> #<CLOSURE (FLET QUICKLISP-CLIENT::QL :IN QUICKLISP-CLIENT:QUICKLOAD) {10040F271B}>) [fast-method]
24: ((:METHOD QL-IMPL-UTIL::%CALL-WITH-QUIET-COMPILATION :AROUND (QL-IMPL:SBCL T)) #<QL-IMPL:SBCL {10053D7553}> #<CLOSURE (FLET QUICKLISP-CLIENT::QL :IN QUICKLISP-CLIENT:QUICKLOAD) {10040F271B}>) [fast-method]
25: ((:METHOD QUICKLISP-CLIENT:QUICKLOAD (T)) "lem" :PROMPT NIL :SILENT NIL :VERBOSE NIL) [fast-method]
26: (QL-DIST::CALL-WITH-CONSISTENT-DISTS #<CLOSURE (LAMBDA NIL :IN QUICKLISP-CLIENT:QUICKLOAD) {10040E997B}>)
27: (INSTALL-SYSTEM-SCRIPT "lem")
28: (INSTALL-SYSTEM-IF-PROBED "lem")
29: (INSTALL NIL)
30: (SB-INT:SIMPLE-EVAL-IN-LEXENV (APPLY (QUOTE MAIN) ROSWELL:*ARGV*) #<NULL-LEXENV>)
31: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) #<NULL-LEXENV>)
32: (SB-EXT:EVAL-TLF (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL NIL)
33: ((LABELS SB-FASL::EVAL-FORM :IN SB-INT:LOAD-AS-SOURCE) (ROSWELL:QUIT (APPLY (QUOTE MAIN) ROSWELL:*ARGV*)) NIL)
34: (SB-INT:LOAD-AS-SOURCE #<CONCATENATED-STREAM :STREAMS NIL {100339C463}> :VERBOSE NIL :PRINT NIL :CONTEXT "loading")
35: ((FLET SB-FASL::THUNK :IN LOAD))
36: (SB-FASL::CALL-WITH-LOAD-BINDINGS #<CLOSURE (FLET SB-FASL::THUNK :IN LOAD) {7F7B0C02753B}> #<CONCATENATED-STREAM :STREAMS NIL {100339C463}>)
37: ((FLET SB-FASL::LOAD-STREAM :IN LOAD) #<CONCATENATED-STREAM :STREAMS NIL {100339C463}> NIL)
38: (LOAD #<CONCATENATED-STREAM :STREAMS NIL {100339C463}> :VERBOSE NIL :PRINT NIL :IF-DOES-NOT-EXIST T :EXTERNAL-FORMAT :DEFAULT)
39: ((FLET ROSWELL::BODY :IN ROSWELL:SCRIPT) #<SB-SYS:FD-STREAM for "file /etc/roswell/install.ros" {1003397873}>)
40: (ROSWELL:SCRIPT "/etc/roswell/install.ros" "cxxxr/lem")
41: (ROSWELL:RUN ((:EVAL "(ros:quicklisp)") (:SCRIPT "/etc/roswell/install.ros" "cxxxr/lem") (:QUIT NIL)))
42: (SB-INT:SIMPLE-EVAL-IN-LEXENV (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/etc/roswell/install.ros" "cxxxr/lem") (:QUIT NIL)))) #<NULL-LEXENV>)
43: (EVAL (ROSWELL:RUN (QUOTE ((:EVAL "(ros:quicklisp)") (:SCRIPT "/etc/roswell/install.ros" "cxxxr/lem") (:QUIT NIL)))))
44: (SB-IMPL::PROCESS-EVAL/LOAD-OPTIONS ((:EVAL . "(progn #-ros.init(cl:load \"/etc/roswell/init.lisp\"))") (:EVAL . "(ros:run '((:eval\"(ros:quicklisp)\")(:script \"/etc/roswell/install.ros\"\"cxxxr/lem\")(:quit ())))")))
45: (SB-IMPL::TOPLEVEL-INIT)
46: ((FLET SB-UNIX::BODY :IN SB-EXT:SAVE-LISP-AND-DIE))
47: ((FLET "WITHOUT-INTERRUPTS-BODY-27" :IN SB-EXT:SAVE-LISP-AND-DIE))
48: ((LABELS SB-IMPL::RESTART-LISP :IN SB-EXT:SAVE-LISP-AND-DIE))

unhandled condition in --disable-debugger mode, quitting
; 
; compilation unit aborted
;   caught 1 fatal ERROR condition

Details of setup:
Debian 9.4
SBCL 1.3.14.debian
roswell 18.8.10.93
ncurses5-config 6.0.20161126

I am very new to lisp and usually get some hints from compile/install messages but this time this is over my head.

Any suggestions on how to resolve this?  

Thanks.

**Comment by fukamachi** [comment]
I suppose there's uncompatible SLIME and ASDF are loaded.
As SBCL 1.3.14 is fairly old, it doesn't surprise me.

Just uninstall the system-wide SBCL and use one which was installed via Roswell.

**Comment by aykaramba** [comment]
Thank you very much for the tip.  I had to upgrade sbcl and asdf as well.  For any newbs reading this on a Debian 9 system:

1) Delete .roswell folder to start fresh after each try.
2) You will need ncurses installed.
3) Force roswell to install a local newer copy of sbcl that it will use with: ros run
4) I had to get roswell to install asdf as the default one in Debian 9 was too old with: ros install asdf
5) Then installing lem worked with:  ros install cxxxr/lem

I appreciate all the help.  This is now resolved.

Robert
